### PR TITLE
Qt/Debugger: Memory View font based sizing, row coloring

### PR DESCRIPTION
**Pull request by TryTwo**
Allows more rows to fit on screen.
Added alternating row colors. I can remove it if disliked.
Added a mark to always indicate the middle (address focused) row. I didn't want to mark the whole row, so the BP box is just darker grey.

### CODE REVIEW

File: Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp [hunk 1]
```diff
@@ -32,8 +32,14 @@ MemoryViewWidget::MemoryViewWidget(QWidget* parent) : QTableWidget(parent)
...
   setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
   setShowGrid(false);
+  setAlternatingRowColors(true);
 
   setFont(Settings::Instance().GetDebugFont());
+  QFontMetrics fm(Settings::Instance().GetDebugFont());
+
+  // Row height as function of text size. Less height than default.
+  const int fonth = fm.height() + 3;
+  verticalHeader()->setMaximumSectionSize(fonth);
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp]: Why is this `fm.height() + 3` when #7697 only uses `fm.height() + 1` for `verticalHeader()->setMaximumSectionSize(...)`?

**Comment by TryTwo** [comment]
memoryview

File: Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp (hunk 1 above)
Comment (TryTwo) [reply Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp]: Memory View being a series of numbers is too cramped and hard to decipher with a very tight row height. Code view looks okay to me.

**Comment by TryTwo** [comment]
Oops, forgot to change the Update loop using a constant row height. Also fonth -> rowh.

**Comment by TryTwo** [comment]
And forgot to set variable BP column/icon size. Should be all done now.

File: Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp [hunk 2]
```diff
@@ -185,19 +194,23 @@ void MemoryViewWidget::Update()
 
     if (row_breakpoint)
     {
-      bp_item->setData(Qt::DecorationRole,
-                       Resources::GetScaledThemeIcon("debugger_breakpoint").pixmap(QSize(24, 24)));
+      bp_item->setData(
+          Qt::DecorationRole,
+          Resources::GetScaledThemeIcon("debugger_breakpoint").pixmap(QSize(rowh, rowh)));
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp]: Is it a good idea to scale this icon?

File: Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp [hunk 3]
```diff
@@ -32,8 +32,14 @@ MemoryViewWidget::MemoryViewWidget(QWidget* parent) : QTableWidget(parent)
   verticalHeader()->hide();
   setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
   setShowGrid(false);
+  setAlternatingRowColors(true);
 
   setFont(Settings::Instance().GetDebugFont());
+  QFontMetrics fm(Settings::Instance().GetDebugFont());
```
Comment (lioncash) [inline Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp]:
```suggestion
  const QFontMetrics fm(Settings::Instance().GetDebugFont());
```

File: Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp [hunk 4]
```diff
@@ -72,7 +78,10 @@ void MemoryViewWidget::Update()
   if (rowCount() == 0)
     setRowCount(1);
 
-  setRowHeight(0, 24);
+  QFontMetrics fm(Settings::Instance().GetDebugFont());
```
Comment (lioncash) [inline Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp]:
```suggestion
  const QFontMetrics fm(Settings::Instance().GetDebugFont());
```

**Comment by TryTwo** [comment]
I noticed this doesn't work as well as code view's font based sizing, because it doesn't re-calibrate on font change/signal. I'll have to update it.

**Comment by TryTwo** [comment]
Moved the verticalheader stuff into Update and tweaked it.
Run Update on font change. 
Added horizontalHeader()->setMinimumSectionSize to let BP column be smaller.

verticalHeader()->setDefaultSectionSize  appears to do what setRowHeight was doing just fine, so removed setRowHeight..

**Pull request closed without merging**